### PR TITLE
[Merged by Bors] - feat(GroupTheory/ArchimedeanDensely): linear ordered group subsets are WF if discrete

### PR DESCRIPTION
**Pull request by pechersky**
---

- [x] depends on: #18479
- [x] depends on: #18480

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/ArchimedeanDensely.lean [hunk 1]
```diff
@@ -297,3 +298,135 @@ lemma LinearOrderedCommGroupWithZero.discrete_iff_not_denselyOrdered (G : Type*)
       MulEquiv.trans_apply, MulEquiv.coe_mk, Equiv.coe_fn_symm_mk, Equiv.coe_fn_mk]
     split_ifs <;>
     simp_all [← Units.val_le_val]
+
+ section WellFounded
```
Comment (acmepjz) [inline Mathlib/GroupTheory/ArchimedeanDensely.lean:302]:
```suggestion
section WellFounded
```

File: Mathlib/GroupTheory/ArchimedeanDensely.lean [hunk 2]
```diff
@@ -297,3 +298,135 @@ lemma LinearOrderedCommGroupWithZero.discrete_iff_not_denselyOrdered (G : Type*)
...
+      simp [WithZero.withZeroUnitsEquiv, MulEquiv.unzero]
+  rw [← Set.wellFoundedOn_sdiff_singleton (a := 0)]
+  refine ⟨fun h ↦ (h.mapsTo Units.val ?_).mono' ?_,
+    fun h ↦ (h.mapsTo ?_ ?_).mono' ?_⟩
+  · intro
+    simp [← Units.val_le_val]
+  · simp [Function.onFun]
+  · exact fun x ↦ if h : x = 0 then 1 else Units.mk0 x h
+  · intro
+    simp (config := {contextual := true}) [← Units.val_le_val]
```
Comment (jcommelin) [inline Mathlib/GroupTheory/ArchimedeanDensely.lean:397]:
```suggestion
    simp +contextual [← Units.val_le_val]
```
Do you evben need the intro, if you do a contextual simp?

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/GroupTheory/ArchimedeanDensely.lean (hunk 2 above)
Comment (pechersky) [reply Mathlib/GroupTheory/ArchimedeanDensely.lean:397]: Yes, needed if to break open `MapsTo`. Instead, I'll add `MapsTo` so the local simp set

**Pull request closed without merging**